manual/gce-sru-20.2.45.txt

```diff
@@ -0,0 +1,1857 @@
+++ value=setup_proposed.sh
++ '[' -z setup_proposed.sh -o setup_proposed.sh = UNSET ']'
++ for var in $@
++ eval 'value=$SAMPLE_CLOUDCONFIG > /dev/null'
+++ value=sethostname.yaml
++ '[' -z sethostname.yaml -o sethostname.yaml = UNSET ']'
++ for var in $@
++ eval 'value=$SSHOPTS > /dev/null'
+++ value=-o
++ '[' -z -o -o -o = UNSET ']'
++ for var in $@
++ eval 'value=$GCE_ZONE > /dev/null'
+++ value=us-central1-a
++ '[' -z us-central1-a -o us-central1-a = UNSET ']'
++ for var in $@
++ eval 'value=$NETCFG_FILE > /dev/null'
+++ value=/etc/network/interfaces.d/50-cloud-init.cfg
++ '[' -z /etc/network/interfaces.d/50-cloud-init.cfg -o /etc/network/interfaces.d/50-cloud-init.cfg = UNSET ']'
++ create_samplecloudconfig SAMPLE_CLOUDCONFIG
++ filename=SAMPLE_CLOUDCONFIG
++ cat
++ create_setup_proposed_script
++ '[' 0 -eq 0 ']'
++ cat
++ echo '### BEGIN xenial'
+### BEGIN xenial
++ case $UBUNTU_RELEASE in
++ family=ubuntu-1604-lts
++ gcloud compute instances create xenial-sru-test --image-family ubuntu-1604-lts --image-project ubuntu-os-cloud-devel --metadata-from-file user-data=sethostname.yaml --zone=us-central1-a
+Created [https://www.googleapis.com/compute/v1/projects/ubuntu-server-277015/zones/us-central1-a/instances/xenial-sru-test].
+NAME             ZONE           MACHINE_TYPE   PREEMPTIBLE  INTERNAL_IP  EXTERNAL_IP     STATUS
+xenial-sru-test  us-central1-a  n1-standard-1               10.128.0.22  35.223.168.228  RUNNING
+++ gcloud compute instances list
+++ grep xenial-sru
+++ awk '{printf "%s", $5}'
++ VM_IP=35.223.168.228
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@35.223.168.228 -- cloud-init status --wait --long
+...............................................................................................................................................................................................................................................................
+status: done
+time: Tue, 23 Jun 2020 23:51:44 +0000
+detail:
+DataSourceGCE
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@35.223.168.228 -- dpkg-query --show cloud-init
+cloud-init	19.4-33-gbb4131a2-0ubuntu1~16.04.1
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@35.223.168.228 -- sudo cat /run/cloud-init/result.json
+sudo: unable to resolve host SRU-worked-gce
+{
+ "v1": {
+  "datasource": "DataSourceGCE",
+  "errors": []
+ }
+}
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@35.223.168.228 -- '!' grep Trace /var/log/cloud-init.log
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@35.223.168.228 -- systemd-analyze
+Startup finished in 2.642s (kernel) + 1min 53.482s (userspace) = 1min 56.124s
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@35.223.168.228 -- systemd-analyze blame
+     1min 7.272s cloud-config.service
```

Comment:
I don't know, but that seems like an odd time for cloud-config.service

Reply:
@lucasmoura @TheRealFalcon this is a good catch, and if you look around line 2070 of the visual diff (cloud-init analyze blame) you can see that that ssh-import-id was the true cost of this
 |`->config-ssh-import-id ran successfully @42.79300s +65.61700s   (1 minute and 5 seconds added in order to ssh-import-id)

So, this points to an intermittent network issue on the system. But, not something that I'd be concerned about with this verification.